web/controllers/user_controller.ex

```diff
@@ -14,27 +15,24 @@ defmodule CodeCorps.UserController do
   end
 
   def handle_create(conn, attributes) do
-    %User{}
-    |> User.registration_changeset(attributes)
-    |> Repo.insert
-    |> login(conn)
-    |> track_signup
+    with {:ok, user} <- %User{} |> User.registration_changeset(attributes) |> Repo.insert,
+         conn        <- login(user, conn)
+    do
+      CodeCorps.Analytics.Segment.track({:ok, user}, :signed_up, conn)
+    else
+      {:error, changeset} -> changeset
+    end
   end
 
-  defp login({:error, changeset}, conn), do: {:error, changeset, conn}
-  defp login({:ok, model}, conn) do
-    {:ok, model, conn |> Plug.Conn.assign(:current_user, model)}
-  end
-
-  defp track_signup({status, model_or_changeset, conn}) do
-    CodeCorps.Analytics.Segment.track({status, model_or_changeset}, :signed_up, conn)
-  end
+  defp login(user, conn), do: Plug.Conn.assign(conn, :current_user, user)
 
-  def handle_update(conn, model, attributes) do
-    model
-    |> User.update_changeset(attributes)
-    |> Repo.update
-    |> CodeCorps.Analytics.Segment.track(:updated_profile, conn)
+  def handle_update(conn, record, attributes) do
+    with {:ok, user, _platform_customer_updates, _connect_customer_updates} <- UserService.update(record, attributes)
```

Comment:
I would just go unnamed on the nil returns here, straight `_`. See no harm in that and easier to read.